### PR TITLE
Data migration

### PR DESCRIPTION
**Pull request by yangyansong-adbe**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: AEPUserProfile/Sources/UserProfile.swift [hunk 1]
```diff
@@ -198,3 +204,21 @@ public class UserProfile: NSObject, Extension {
         }
     }
 }
+
+public enum UserProfileV5Migrator {
```
Comment (shalehaha) [inline AEPUserProfile/Sources/UserProfile.swift:208]: shouldn't be public

File: AEPUserProfile/Sources/UserProfile.swift [hunk 2]
```diff
@@ -198,3 +204,21 @@ public class UserProfile: NSObject, Extension {
...
+    static func existingAttributes() -> [String: Any]? {
+        guard let json = UserDefaults.standard.object(forKey: UserProfileConstants.V5Migration.USER_PROFILE_KEY) as? String, let jsonData = json.data(using: .utf8) else {
```
Comment (shalehaha) [inline AEPUserProfile/Sources/UserProfile.swift:210]: it'd better to first check if `UserProfileConstants.V5Migration.USER_PROFILE_KEY` exists, if not, print a trace log saying not data to migrate.

File: AEPUserProfile/Sources/UserProfile.swift [hunk 3]
```diff
@@ -49,6 +49,12 @@ public class UserProfile: NSObject, Extension {
         registerListener(type: EventType.userProfile, source: EventSource.requestReset, listener: removeAttributes(event:))
         registerListener(type: EventType.rulesEngine, source: EventSource.responseContent, listener: handleRulesEngineResponse(event:))
         loadAttributes()
+        if let existingAttributes = UserProfileV5Migrator.existingAttributes() {
+            Log.trace(label: UserProfile.LOG_TAG, "Start to do data migration")
```
Comment (shalehaha) [inline AEPUserProfile/Sources/UserProfile.swift:53]: "Start data migration"

File: AEPUserProfile/Sources/UserProfile.swift (hunk 1 above)
Comment (shalehaha) [reply AEPUserProfile/Sources/UserProfile.swift:208]: also I'd move it to a sperate class

**Review by shalehaha — CHANGES_REQUESTED** [review]

**Comment by yangyansong-adbe** [comment]
@shalehaha  Review comments have been addressed.

**Review by shalehaha — APPROVED** [review]

**Pull request merged**